src/geom/CoordinateSequence.cpp

```diff
@@ -288,7 +288,7 @@ CoordinateSequence::getOrdinate(std::size_t index, std::size_t ordinateIndex) co
         case CoordinateSequence::Z:
             return hasZ() ? getAt<Coordinate>(index).z : DoubleNotANumber;
         case CoordinateSequence::M:
-            return hasM() ? (hasZ() ? getAt<CoordinateXYZM>(index).m : getAt<CoordinateXYM>(index).m) : DoubleNotANumber;
+            return hasM() ? getAt<CoordinateXYZM>(index).m : DoubleNotANumber;
```

Comment:
This works, as there seems to be an issue with `CoordinateXYM`.

Reply:
@dbaston if you want to investigate the issue, re-set this line to the previous edit to see the issue via `ctest --output-on-failure -R GEOSGeomGetXYZM`. If the issue is simple, feel free to push a commit here on in another PR/main.

Reply:
In current GEOS, `POINT M (1 2 3)` is backed by `CoordinateXYZM`. So to check if we can access the coordinate as an XYZM or an XYM we need to ask that question directly, not infer it from hasZ/hasM.